util/minetestmapper.py

```diff
@@ -190,8 +190,77 @@ def usage():
         draworigin = True
     elif o == "--drawunderground":
         drawunderground = True
+    elif o == "--geometry":
+        geometry_string = a
+    elif o == "--region":
+        region_string = a
     else:
         assert False, "unhandled option"
+nonchunky_xmin=-1500
+nonchunky_xmax=1500
+nonchunky_zmin=-1500
+nonchunky_zmax=1500
+
+if geometry_string is not None:
+    parts = geometry_string.split("+")
+    if len(parts) == 3:
+        coords_string, width_string, height_string = parts
+        coords = coords_string.split(":")
+        if len(coords)==2:
+            x_string, z_string = coords
+            x = int(x_string)
+            z = int(z_string)
+            this_width = int(width_string)
+            this_height = int(height_string)
+            nonchunky_xmin = x
+            nonchunky_xmax = nonchunky_xmin + this_width - 1  # -1 since max is inclusive rect
+            nonchunky_zmin = z
+            nonchunky_zmax = nonchunky_zmin + this_height - 1  # -1 since max is inclusive rect
+            print("#geometry:")
+            print("#  x:"+str(x))
+            print("#  z:"+str(z))
+            print("#  width:"+str(this_width))
+            print("#  height:"+str(this_height))
+            print("region:")
+            print("  xmin:"+str(nonchunky_xmin))
+            print("  xmax:"+str(nonchunky_xmax))
+            print("  zmin:"+str(nonchunky_zmin))
+            print("  zmax:"+str(nonchunky_zmax))
+        else:
+            print("ERROR: (Missing coordinates in '"+geometry_string+"' for geometry) Geometry should be in the form: x:z+width+height")
```

Comment:
This line looks a little too long, for PEP8 compliance it needs to be <= 80 columns.